snuba/datasets/schema.py

```diff
@@ -75,11 +70,59 @@ def get_column_differences(self, expected_columns: Mapping[str, str]) -> List[st
 
     def get_migration_statements(
         self
-    ) -> Callable[[str, Mapping[str, str]], Sequence[str]]:
+    ) -> Optional[Callable[[str, Mapping[str, str]], Sequence[str]]]:
         return self.__migration_function
 
 
 class TableSchema(Schema):
+    """
+    Represent a table-like schema. This means it represents either
+    a Clickhouse table, a Clickhouse view or a Materialized view.
+
+    Going on, this abstraction will be split into an abstract entity
+    and a Clickhouse specific TableStore on which we can write.
+    """
+
+    TEST_TABLE_PREFIX = "test_"
+
+    def __init__(self,
+        local_table_name: str,
+        dist_table_name: str,
+        columns: ColumnSet,
+        migration_function: Optional[Callable[[str, Mapping[str, str]], Sequence[str]]]=None,
```

Comment:
It seems like this argument order should be made consistent with the superclass (`columns` first.) The optional `migration_function` makes that difficult to apply consistently, so I wonder if everything else should be explicitly keyword-only arguments like we did for datasets?

Reply:
right, will fix